misc/python/materialize/cli/mzexplore.py

```diff
@@ -0,0 +1,379 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import json
+import string
+import textwrap
+from contextlib import closing
+from pathlib import Path
+from typing import Optional
+
+import click
+import pandas as pd
+
+from ..mzexplore import ExplainFlag, ExplainStage, sql
+
+# import logging
+# logging.basicConfig(encoding='utf-8', level=logging.DEBUG)
+
+# Click CLI Application
+# ---------------------
+
+
+@click.group()
+def app() -> None:
+    pass
+
+
+class Arg:
+    repository = dict(
```

Comment:
Because the definitions of these key value pairs are then closer to the syntax [advertised by Click](https://click.palletsprojects.com/en/8.1.x/options/#multiple-options). For example:

```python
@click.option('--message', '-m', multiple=True)
```

and the factored out common part of the argument is
```python
dict(multiple=True)
```
and then I do
```python
@click.option('--message', **Options.MY_OPTION)
```